docs/src/main/asciidoc/rest-client-multipart-guide.adoc

```diff
@@ -0,0 +1,325 @@
+We are assuming the payload is well-known before the request is sent, so we can model as a POJO.
+
+[NOTE]
+====
+If the payload is unknown, you can also use the RESTEasy custom API instead. If that's the case, see the RESTEasy Multipart Providers link in the end of the guide.
+====
+
+Our first order of business is to setup the model we will be using to define the `multipart/form-data` payload, in the form of a `MultipartBody` POJO.
+
+Create a `src/main/java/org/acme/restclient/multipart/MultipartBody.java` file and set the following content:
+
+[source,java]
+----
+package org.acme.restclient.multipart;
+
+import java.io.InputStream;
+
+import javax.ws.rs.FormParam;
+import javax.ws.rs.core.MediaType;
+
+import org.jboss.resteasy.annotations.providers.multipart.PartType;
+
+public class MultipartBody {
+
+    @FormParam("file")
+    @PartType(MediaType.APPLICATION_OCTET_STREAM)
+    private InputStream file;
+
+    @FormParam("fileName")
+    @PartType(MediaType.TEXT_PLAIN)
+    private String name;
+
+    public void setFile(InputStream file) {
+        this.file = file;
+    }
+
+    public InputStream getFile() {
+        return file;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public String getName() {
+        return name;
+    }
+}
+----
+
+The purpose of the annotations in the code above is the following:
+
+* `@FormParam` is a standard JAX-RS annotation used to define a form parameter contained within a request entity body
+* `@PartType` is a RESTEasy specific annotation required when a client performs a multipart request and defines the content type for the part.
+
+== Create the interface
+
+Using the MicroProfile REST Client is as simple as creating an interface using the proper JAX-RS and MicroProfile annotations. In our case the interface should be created at `src/main/java/org/acme/restclient/multipart/MultipartService.java` and have the following content:
+
+[source, java]
+----
+package org.acme.restclient.multipart;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;
+import org.jboss.resteasy.annotations.providers.multipart.MultipartForm;
+
+@Path("/echo")
+@RegisterRestClient
+public interface MultipartService {
+
+    @POST
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    @Produces(MediaType.TEXT_PLAIN)
+    String sendMultipartData(@MultipartForm MultipartBody data);
+
+}
+----
+
+The `sendMultipartData` method gives our code the ability to POST a `multipart/form-data` request to our Echo service (running in the same server for demo purposes).
+Because in this demo we have a exact knowledge of the `multipart/form-data` packets, we can map them to the model class created in the previous section using the `@org.jboss.resteasy.annotations.providers.multipart.MultipartForm` annotation.
```

Comment:
> Because in this demo we have a exact knowledge
...**the** exact knowledge?

Reply:
This excerpt was taken from the RESTEasy docs -[ 26.11. @MultipartForm and POJOs](https://docs.jboss.org/resteasy/docs/4.3.1.Final/userguide/html/Multipart.html)
, but I'm fine to change it